pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -544,6 +557,55 @@ private void doAddConsumingSegment(String segmentName)
     _logger.info("Added new CONSUMING segment: {}", segmentName);
   }
 
+  @Override
+  public File downloadSegment(SegmentZKMetadata zkMetadata)
+      throws Exception {
+    if (!PauselessConsumptionUtils.isPauselessEnabled(_tableConfig)) {
```

Comment:
Let's rely on segment status instead of table config to decide how to download the segment. There is no guarantee that the table config is in sync with the segment (e.g. user changes table config in the middle)